src/ambianic/webapp/flaskr.py

```diff
@@ -159,6 +162,29 @@ def get_status():
         resp = jsonify(response_object)
         return resp
 
+    @app.route('/api/auth/premium-notification')
+    def initialize_premium_notification():
+        userAuth0Id = request.args.get('userId')
+        endpoint = request.args.get('notification_endpoint')
+        auth_file = {
+            'name': 'AMBIANIC-EDGE-PREMIUM',
+            'credentials': {
+                'USER_AUTH0_ID': userAuth0Id,
+                'NOTIFICATION_ENDPOINT': endpoint,
+            }}
+
+        directory = pkg_resources.resource_filename(
+            "ambianic.webapp", "premium.yaml")
+
+        try:
+            file = open(directory, 'w+')
+            yaml.dump(auth_file, file)
+            file.close()
+        except FileNotFoundError:
+            log.debug("Error locating file")
```

Comment:
@vickywane same comment here. Do we need a warn or error log level? Is debug sufficient?

Reply:
I think `warn` is sufficent. 

I however can see the `debug` level being used in other parts of the codebase.

Reply:
>  how did you address this open thread?

Are you making reference to the file opened using the `with` statement?

If yes, 
It is automatically closed and I do not need to manually close it as I using the `with` statement not `open`.

[This](https://lerner.co.il/2015/01/18/dont-use-python-close-files-answer-depends/) article contains more explanation on this.